Automattic/jetpack
# My Jetpack: add A/B experiment to welcome flow

Context: pbxNRc-3UJ-p2

## Proposed changes:

* This PR adds an A/B experiment to new welcome flow. In order to load the assignment right after site-connecting asynchronously, a new function was added to load w.js synchronously, so it initializes the cookie necessary to load the assignment. This function was added to `side-load-tracks.ts`.
For more context about the experiment, see the P2 post linked above. 

### Other information:


## Jetpack product discussion


pbxNRc-3UJ-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Spin up a JN site with Jetpack Beta pointing to this branch and with sandbox access enabled
* Visit your site, go to Settings -> Jetpack Constants on `/wp-admin/options-general.php?page=companion_settings` and point JETPACK__SANDBOX_DOMAIN to your WPCOM sandbox
* Go to My Jetpack and open your browser's developer console on the network tab
* Click on "Activate Jetpack in one click"
<img width="1101" alt="image" src="https://github.com/user-attachments/assets/b5209b95-641d-49f1-a742-baded6fefe65">

* It'll then redirect you to the user-connection screen
* Filter your network request by `assignment` and confirm you see a response like this:

```json
{
    "variations": {
        "jetpack_my_jetpack_post_connection_flow_202408": null
    },
    "ttl": 60,
    "debug": {
        "backend_aa_result": "request not sampled"
    }
}
```

* To test the treatment variation, disconnect your site by going back to My Jetpack, scrolling down to "Connection", clicking on "Manage" and then disconnecting; [new] then go to "Dashboard", scroll down and click "Reset Options (dev only)" in the footer.
<img width="547" alt="image" src="https://github.com/user-attachments/assets/8bc481f0-5632-4cd7-8846-90240673c7c7"> 

![CleanShot 2024-08-05 at 11 41 31@2x](https://github.com/user-attachments/assets/336de814-da3f-4e62-855d-593a622f6326)


* Then, follow the instructions on this page 21912-explat-experiment, hovering "Bookmarklet" for treatment in the Audience section
